.github/workflows/benchmark-on-label.yml

```
@@ -0,0 +1,123 @@
+          fetch-depth: 1
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5
+        with:
+          python-version: "3.10"
+          cache: "pip"
+
+      - name: Install dependencies
+        working-directory: valkey-perf-benchmark
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+          pip install -r requirements.txt
+
+      - name: Run benchmarks
+        timeout-minutes: 60
+        working-directory: valkey-perf-benchmark
+        run: |
+          CONFIG_FILE="../valkey/.github/benchmark_configs/pr_benchmark.json"
+
+          # Base benchmark arguments
+          BENCHMARK_ARGS=(
+            --commits "${{ github.event.pull_request.merge_commit_sha }}"
+            --baseline "${{ github.event.pull_request.base.ref }}"
+            --results-dir "results"
+            --valkey-path "../valkey"
+            --config "$CONFIG_FILE"
+          )
+
+          # Add server and client CPU ranges for official repository as we are running on a metal instance
+          if [[ "${{ github.repository }}" == "valkey-io/valkey" ]]; then
+            BENCHMARK_ARGS+=(--server-cpu-range 92-93)
+            BENCHMARK_ARGS+=(--client-cpu-range 94-95)
+          fi
+
+          # Run benchmark
+          python ./benchmark.py "${BENCHMARK_ARGS[@]}"
+
+      - name: Compare results
+        working-directory: valkey-perf-benchmark
+        run: |
+          python ./utils/compare_benchmark_results.py \
+            ./results/${{ github.event.pull_request.base.ref }}/metrics.json \
+            ./results/${{ github.event.pull_request.merge_commit_sha }}/metrics.json \
+            ../comparison.md
+
+      - name: Upload artifacts
+        if: always()
+        continue-on-error: true
+        uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
+        with:
+          name: benchmark-results
+          path: |
+            ./valkey-perf-benchmark/results/${{ github.event.pull_request.merge_commit_sha }}/metrics.json
+            ./valkey-perf-benchmark/results/${{ github.event.pull_request.base.ref }}/metrics.json
+            comparison.md
+
+      - name: Comment PR with results
+        uses: actions/github-script@5c56fde4671bc2d3592fb0f2c5b5bab9ddae03b1 # v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs');
+            const body = fs.readFileSync('comparison.md', 'utf8');
+            await github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body
+            });
+
+      - name: Remove ${{ github.event.label.name }} label
+        if: always() && (github.event.label.name == 'run-benchmark' || github.event.label.name == 'run-cluster-benchmark')
```

We don't need the `run-cluster-benchmark` label further. What does `always()` do ?

```suggestion
        if: always() && github.event.label.name == 'run-benchmark'
```